### ISSUE TITLE
Warning: Undefined array key 1 in \Date::resolveCustomModifiers

### ISSUE THREAD

**Issue by BugBuster1701** [contao/contao#2865]
**Affected version(s)**

- Contao 4.11 (4.9 is certainly also affected.)
- PHP 8
- Debug Modus

**Description**

When in my Backend Module:
`\Date::parse("D, d.m.Y", 1615330800);`
is called,the output results:

```
ErrorException:
Warning: Undefined array key 1

  at vendor/contao/core-bundle/src/Resources/contao/library/Contao/Date.php:641
  at Contao\Date::resolveCustomModifiers()
     (vendor/contao/core-bundle/src/Resources/contao/library/Contao/Date.php:579)
  at Contao\Date::parse()
...
```

- resolveCustomModifiers($strDate) , the parameter is here: "3::2, 10.03.2021"
- $chunks in row 637 is 
```
array:3 [
  0 => ""
  1 => "3::2"
  2 => ", 10.03.2021"
]
```
The Problem is then here, when $chunk = "" : 
```
foreach ($chunks as $chunk)
{
	list($index, $flag) = explode('::', $chunk);
```

**Comment by leofeyer** [comment]
Can you please test if replacing `PREG_SPLIT_DELIM_CAPTURE` with `PREG_SPLIT_DELIM_CAPTURE|PREG_SPLIT_NO_EMPTY` solves the issue for you?

**Comment by BugBuster1701** [comment]
No, not a solution.
The chunk Array is now:
```
array:2 [▼
  0 => "4::2"
  1 => ", 18.03.2021"
]
```
The first empty string is removed, but ", 18.03.2021" is the problem now.

My workaround was:
```
	foreach ($chunks as $chunk)
	{
		if (empty($chunk) || false === strpos($chunk,'::')) { $strReturn .= $chunk; continue; }

		list($index, $flag) = explode('::', $chunk);
```
Combined with your bugfix:
```
	$chunks = preg_split("/([0-9]{1,2}::[1-4])/", $strDate, -1, PREG_SPLIT_DELIM_CAPTURE|PREG_SPLIT_NO_EMPTY);

	foreach ($chunks as $chunk)
	{
		if (false === strpos($chunk,'::')) { $strReturn .= $chunk; continue; }

		list($index, $flag) = explode('::', $chunk);
```

**Comment by leofeyer** [comment]
Ok, then please try replacing `list($index, $flag) = explode('::', $chunk);` with `list($index, $flag) = explode('::', $chunk) + array(null, null);`. Does it fix the issue?

**Comment by BugBuster1701** [comment]
Cool, it works. With and without PREG_SPLIT_NO_EMPTY.

**Comment by leofeyer** [comment]
Fixed in 16ece13aa81de8c5a30c0c07bc557e847d108712.
